src/Client.php

```diff
@@ -703,6 +703,18 @@ public function setHostname($hostname)
         return $this;
     }
 
+    /**
+    * Adds new data fields to the device data collection.
+    *
+    * @param array $data an associative array containing the new data to be added
+    *
+    * @return this
+    */
+    public function mergeDeviceData($data)
```

Comment:
Forgot one thing: This isn't needed here. For public usage, `$client->getConfig()->mergeDeviceData()` is available. Older methods were copied to avoid making a breaking change during #510, but since this is new, there's no need to have duplicate methods.